eth/api.go

```diff
@@ -652,18 +652,21 @@ func (s *PublicBlockChainAPI) doCall(args CallArgs, blockNr rpc.BlockNumber) (st
 	// Assemble the CALL invocation
 	msg := callmsg{
 		from:     from,
-		to:       &args.To,
 		gas:      args.Gas.BigInt(),
 		gasPrice: args.GasPrice.BigInt(),
 		value:    args.Value.BigInt(),
 		data:     common.FromHex(args.Data),
 	}
+	if args.To != (common.Address{}) {
```

Comment:
We must compare to `nil`. Checking if `to != common.Address{}` would field false if someone actually send funds to the zero'd address. `if args.To != (common.Address{}) {` is therefor incorrect.